src/main/java/org/springframework/social/salesforce/api/Salesforce.java

```diff
@@ -39,6 +39,8 @@
 
     public UserOperations userOperations();
 
+    public UserOperations userOperations(String gatewayUrl);
```

Comment:
Do we need this?  It looks like it could lead to confusing behavior, i.e. it sets the gatewayUrl used for all future calls rather than just using the override only for this call.

Can we just make  `SalesforceAdapter` call `setAuthGatewayBaseUrl`?